### PR TITLE
feat: new relative period last10financial [DHIS2-7776]

### PR DESCRIPTION
**Pull request by d-bernat**

### CODE REVIEW

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/RelativePeriods.java [hunk 1]
```diff
@@ -922,6 +939,7 @@ public static Period setName( Period period, String periodName, boolean dynamicN
         add( list, RelativePeriodEnum.THIS_FINANCIAL_YEAR, thisFinancialYear );
         add( list, RelativePeriodEnum.LAST_FINANCIAL_YEAR, lastFinancialYear );
         add( list, RelativePeriodEnum.LAST_5_FINANCIAL_YEARS, last5FinancialYears );
+        add( list, RelativePeriodEnum.LAST_10_FINANCIAL_YEARS, last5FinancialYears );
```
Comment (maikelarabori) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/RelativePeriods.java:942]: I believe there is a small mistake: "last5FinancialYears"

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/RelativePeriods.java [hunk 2]
```diff
@@ -712,6 +719,15 @@ public PeriodType getHighestFrequencyPeriodType( List<Period> periods )
                 dynamicNames, format ) );
         }
 
+        if ( isLast10FinancialYears() )
+        {
+            periods.addAll( getRollingRelativePeriodList( financialPeriodType, LAST_10_FINANCIAL_YEARS,
```
Comment (maikelarabori) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/RelativePeriods.java:724]: We are adding to the list twice here. Just wondering if this is expected...

**Review by maikelarabori — APPROVED** [review]
Looks good @d-bernat , just left two small comments to be confirmed by you. :+1: 
Thanks!

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/RelativePeriods.java (hunk 2 above)
Comment (d-bernat) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/RelativePeriods.java:724]: yes, it is. I am not allowed to create a new PeriodType like I did before. 
This is the critical method I have to use in FinancialPeriodType
@Override
    public List<Period> generateRollingPeriods( Date date )
    {
        return generateLast5Years( date );
    }

To extend the method signature means extend the interface, which is breaking change to lead me to lot of changes like I describe before (Last_10_YEARS story). That's why I have to call it twice and manipulate the date param by second call.
Sorry, I didn't found any other solution

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/RelativePeriods.java (hunk 1 above)
Comment (d-bernat) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/RelativePeriods.java]: fixed

**Review by larshelge — APPROVED** [review]

**Pull request merged**